spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/util/DeploymentPropertiesUtils.java

```diff
@@ -83,20 +109,15 @@ private DeploymentPropertiesUtils() {
 				// we skip first as we would not have anything to append to. this
 				// would happen if dep prop string is malformed and first given
 				// key/value pair is not actually a key/value.
-				pairs.set(pairs.size() - 1, pairs.get(pairs.size() - 1) + "," + candidates[i]);
+				pairs.set(pairs.size() - 1, pairs.get(pairs.size() - 1) + delimiter + candidates[i]);
```

Comment:
I've been considering it but this utility is used in couple of places and i am not sure if timing the values this would not break something. That is way the new test also checks that the existing spaces are preserved.